src/v/cluster/health_monitor_frontend.cc

```diff
@@ -147,4 +153,19 @@ ss::future<bool> health_monitor_frontend::does_raft0_have_leader() {
       [](health_monitor_backend& be) { return be.does_raft0_have_leader(); });
 }
 
+ss::future<> health_monitor_frontend::reset() {
+    vlog(clusterlog.info, "Resetting Health Monitor Frontend");
+    // Do not reset _cluster_disk_health because there is no guarantee that disk
+    // space changed.
+    if (ss::this_shard_id() == refresher_shard) {
+        cancel_timer(_refresh_timer);
```

Comment:
nit: Up to consensus if we should cancel and re-arm timers but I don't see the need